src/celeritas/neutron/model/CascadeOptionsIO.json.cc

```diff
@@ -0,0 +1,81 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/model/CascadeOptionsIO.json.cc
+//---------------------------------------------------------------------------//
+#include "CascadeOptionsIO.json.hh"
+
+#include <string>
+#include <nlohmann/json.hpp>
+
+#include "corecel/io/JsonUtils.json.hh"
+
+#include "CascadeOptions.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Read options from JSON.
+ */
+void from_json(nlohmann::json const& j, CascadeOptions& opts)
+{
+#define FDO_INPUT(NAME)                    \
+    do                                     \
+    {                                      \
+        if (j.contains(#NAME))             \
+            j.at(#NAME).get_to(opts.NAME); \
+    } while (0)
```

Comment:
There's a little utility function that makes this easier (and slightly more efficient and slightly more flexible):

```suggestion
#define FDO_INPUT(NAME)  CELER_JSON_LOAD_OPTION(j, opts, NAME)
```
and include "corecel/io/JsonUtils.json.hh"

Reply:
Updated.